app/models/services/base-service.ts

```diff
@@ -204,6 +204,21 @@ export default abstract class BaseService<ServiceParamType> {
     return this.config.capabilities;
   }
 
+  /**
+   * Returns the final staging location of the service.
+   *
+   * @returns the final staging location of the service
+   */
+  finalStagingLocation() : string {
+    const { requestId, destinationUrl } = this.operation;
+    if (destinationUrl) {
+      let destPath = destinationUrl.substring(5);
+      destPath = destPath.endsWith('/') ? destPath.slice(0, -1) : destPath;
+      return defaultObjectStore().getUrlString(destPath, requestId + '/');
+    } 
+    return defaultObjectStore().getUrlString(env.stagingBucket, 'public/' + requestId + '/');
```

Comment:
Nit: style thing - if you find you are adding more than one + to concatenate a string it might be more readable as `public/${requestId}/`. I'm not really sure which is more readable, but figured you'd want to know there's another way too.

Reply:
I like your suggestion better.